### PR TITLE
[front] feat: add buttons to compare criteria

### PR DESCRIPTION
**Pull request by GresilleSiffle**
**related issues** #1940

---

### Description

This PR adds a new comparison method for mobile devices: buttons.

capture

**Features**

- the main criterion is mandatory and can't be skipped
- the arrows allow to navigate between the criteria
- to clear a score given to a criterion, click on the active comparison button
- the criteria are displayed according to the user's preferences

**How the front end chooses to display buttons or sliders?**

If the displayed comparison has been previously made with buttons, the buttons are displayed. If it has been rated by sliders, sliders are displayed. 

If the displayed comparison is new, the button are displayed for devices *not* matching the media query `(pointer:fine)`, and the sliders are displayed for all other devices (See the [pointer](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/pointer) media feature).

No conversion from buttons to sliders and from sliders to buttons is allowed for now.

**Future improvements**

- disable by default the weekly collective goal on mobile devices to save space, and to make the UI more simple, add a setting to control this behaviour
  - addressed by #1977
- *suggested in Discord* add a setting to control display of the buttons auto and select (they are currently displayed above the entity cards, but they could be displayed below to ease the navigation on mobile devices)

**to-do**

*front end - tutorial*
- [x] test the tutorial
- [x] update the texts to match the buttons on mobile devices
 
*front end - buttons*

- [x] make the buttons look like buttons
- [x] make them "selected" when a criterion has been given a score
- [x] disable them during API requests
- [x] add the possibility to swipe up and down
- [x] add hints about swipe up and down
- [x] prevent skipping the main criterion of the poll
- [x] display a "continue or change video" after all mandatory criteria have been rated?
  - not for now

*front end - comparison input*
- [x] display buttons when the `score_max` of the main criterion is 2
- [x] else display sliders if the `score_max` of the main criterion is 10
- [x] else display buttons on smartphones
- [x] else display sliders
- [x] handle incoherent `score_max` between != criteria

*front end - comparison scores review*
- [x] add the possibility to view all criteria at once
  - [x] add a proto of the table view 
- [x] display the review only when buttons are displayed
  - [x] factorise duplicated code 

*front end - comparison*
- [x] add the possibility to clear the score of an already submitted optional criterion
- [x] rework the margin and padding of the grid items to save space

*front end - coherence*

- [x] when displaying a "mobile" comparison on desktop
  - [x] display the buttons, and add a message "this comparison has been made on mobile"
  - [x] allow the users to update their comparison
- [x] when displaying a "desktop" comparison on mobile
  - [x] display the sliders, and add a message "this comparison has been made on desktop"   
  - [x] make the sliders read-only

### Preview

capture

A "mobile" comparison displayed on desktop:

screencapture-localhost-3000-comparison-2024-06-10-11_29_03

### Checklist

- [x] I added the related issue(s) id in the related issues section (if any)
  - if not, delete the related issues section
- [x] I described my changes and my decisions in the PR description
- [x] I read the development guidelines of the [CONTRIBUTING.md][development-guidelines]
- [ ] The tests pass and have been updated if relevant
- [x] The code quality check pass

[development-guidelines]: https://github.com/tournesol-app/tournesol/blob/main/CONTRIBUTING.md#development-guidelines

### CODE REVIEW

**Comment by GresilleSiffle** [comment]
Recent feedback from the rest of the team:
- [ ] try to find an alternative to the icons (+) and (++)
- [ ] try to find an alternative to the icon representing a balance

**Comment by GresilleSiffle** [comment]
Some alternatives to replace the current buttons:

| cap1 | cap2 | cap3 | cap4 |
|---|---|---|---|

**Comment by GresilleSiffle** [comment]
What do you think about the following capture? Instead of displaying the name of the criterion, a full question is asked to increase the UI clarity

capture

**Comment by GresilleSiffle** [comment]
The branch is now ready for review. I keep the PR in draft for now, because the back end is not yet ready to process comparisons with different `score_max` value.

File: frontend/src/features/comparisons/inputs/CriteriaButtons.tsx [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+
+import { useCurrentPoll } from 'src/hooks';
+import { TutorialContext } from 'src/features/comparisonSeries/TutorialContext';
+
+import { ComparisonRequest } from 'src/services/openapi';
+
+import CriterionButtons, { BUTTON_SCORE_MAX } from './CriterionButtons';
+import { useOrderedCriteria } from 'src/hooks/useOrderedCriteria';
+
+const SWIPE_TIMEOUT = 240;
```
Comment (amatissart) [inline frontend/src/features/comparisons/inputs/CriteriaButtons.tsx:17]: I feel the animation is a bit slow and frustrating when looking for a specific criterion. Why not using the same value as in the `EntitySelector`?
```suggestion
const SWIPE_TIMEOUT = 180;
```

Comment (GresilleSiffle) [reply frontend/src/features/comparisons/inputs/CriteriaButtons.tsx]: Addressed by https://github.com/tournesol-app/tournesol/pull/1968/commits/e3bed73721b6e8c20a710419c8174ff4ab3eafa6

**Comment by GresilleSiffle** [comment]
With https://github.com/tournesol-app/tournesol/pull/1968/commits/a9655a13f1a14d17d4e5dc114da8f0ab064a3917 I added an icon in the list of comparisons to indicate whenever a comparison has been made on a mobile device.

capture

**Review by amatissart — APPROVED** [review]

**Pull request merged**